pkg/codegen/dotnet/gen_program_expressions.go

```diff
@@ -407,14 +412,13 @@ func (g *generator) genIntrensic(w io.Writer, from model.Expression, to model.Ty
 			return
 		}
 		var convertFn string
-		switch {
-		case to.Type.Equals(model.StringType):
+		if to.Type.Equals(model.StringType) {
 			convertFn = fmt.Sprintf("System.Enum.Parse<%s.%s>", pkg, name)
-		default:
-			panic(fmt.Sprintf(
-				"Unsafe enum conversions from type %s not implemented yet: %s => %s",
-				from.Type(), from, to))
+		} else {
+			// for numeric enums, we new up the type that is generated for that enum
+			convertFn = fmt.Sprintf("new %s.%s", pkg, name)
```

Comment:
Is there a test that demonstrates this? Do we really use `new`?

Reply:
I looked at the generated and their was a constructor that accepts `double` but now I checked again and that constructor is `private` 😅 
```cs
private TeamStackPermissionScope(double value)
```
there is actually no way to construct the enum from a number (it's generated as a struct). I changed the previous behaviour because it was panicking _too_ early and unnecessarily so I though I would just fix it but this won't do. Will bring back the previous behaviour without redundantly panicking. Also we could change sdk-gen in dotnet to make the constructor public or add an implicit cast operator but that is for another PR.

Reply:
Old behaviour is back without panicking too early ✅ constructing a numeric enum from a number value will be handled in a separate PR